### ISSUE TITLE
CSS Font declarations with # in URL busted incorrectly when rename is false

### ISSUE THREAD

**Issue by jeffsheets** [hollandben/grunt-cache-bust#124]
We are using cacheBust with the rename=false option to append the ? busting param to the URL.

I noticed a couple of small things when busting our concatenated css file that includes twitter bootstrap css. Not a big deal for us, but I just wanted to report it.

Twitter bootstrap defines the Glyphicons font-face using this format:
```
@font-face {
  font-family: 'Glyphicons Halflings';
  src: url('../font/bootstrap/glyphicons-halflings-regular.eot');
  src: url('../font/bootstrap/glyphicons-halflings-regular.eot?#iefix') format('embedded-opentype'),
       url('../font/bootstrap/glyphicons-halflings-regular.woff') format('woff'),
       url('../font/bootstrap/glyphicons-halflings-regular.ttf') format('truetype'),
       url('../font/bootstrap/glyphicons-halflings-regular.svg#glyphicons_halflingsregular') format('svg');
}
```

After running through cacheBust, this is the result:

```
@font-face {
  font-family: 'Glyphicons Halflings';
  src: url('../font/bootstrap/glyphicons-halflings-regular.eot?7e862de50047b996');
  src: url('../font/bootstrap/glyphicons-halflings-regular.eot?7e862de50047b996?#iefix') format('embedded-opentype'),
       url('../font/bootstrap/glyphicons-halflings-regular.woff?6b1bb734715a399d') format('woff'),
       url('../font/bootstrap/glyphicons-halflings-regular.ttf?8d6dcd165c56a9c3') format('truetype'),
       url('../font/bootstrap/glyphicons-halflings-regular.svg#glyphicons_halflingsregular') format('svg');
}
```

2 things:
1- the 2nd .eot reference with '#iefix' gets an extra ?
2- The .svg URL is skipped with a message of
```
Static asset "client\dist\assets\font\bootstrap\glyphicons-halflings-regular.svg#glyphicons_halflingsregular" skipped because it wasn't found, original reference=../font/bootstrap/glyphicons-halflings-regular.svg#glyphicons_halflingsregular
```

I think the result should be similar to:
```
@font-face {
  font-family: 'Glyphicons Halflings';
  src: url('../font/bootstrap/glyphicons-halflings-regular.eot?7e862de50047b996');
  src: url('../font/bootstrap/glyphicons-halflings-regular.eot?7e862de50047b996#iefix') format('embedded-opentype'),
       url('../font/bootstrap/glyphicons-halflings-regular.woff?6b1bb734715a399d') format('woff'),
       url('../font/bootstrap/glyphicons-halflings-regular.ttf?8d6dcd165c56a9c3') format('truetype'),
       url('../font/bootstrap/glyphicons-halflings-regular.svg?4b1bb734715c399a#glyphicons_halflingsregular') format('svg');
}
```

**Comment by hollandben** [comment]
I think this issue came up previous but wasn't reported correctly, so thanks for giving me a full description of what is going on. I will add this as a test case and see if we can get it fixed :)

**Comment by hollandben** [comment]
I've started to fix the issue in #127. It fixes the `../font/bootstrap/glyphicons-halflings-regular.svg?4b1bb734715c399a#glyphicons_halflingsregular` issue but sadly doesn't cover the case with `?#iefix` on the end

**Comment by hollandben** [comment]
Final fix in #130. I don't particularly like the implementation, but will take a longer look at when I have some more time. Have created a reminder in #131

**Comment by jeffsheets** [comment]
Thanks again for the quick response

I pulled in the latest 0.4.13 version. Looks like the .svg one is fixed. But the ?#iefix one is still being busted with two question marks as
```
url('../font/bootstrap/glyphicons-halflings-regular.eot?7e862de50047b9967e862de50047b996?#iefix')
```

One difference I see. The actual css has a long runon line. I had added newlines in my bug report for readability. But the actual unbusted source looks like this:
```
@font-face {
  font-family: 'Glyphicons Halflings';

  src: url('../fonts/glyphicons-halflings-regular.eot');
  src: url('../fonts/glyphicons-halflings-regular.eot?#iefix') format('embedded-opentype'), url('../fonts/glyphicons-halflings-regular.woff') format('woff'), url('../fonts/glyphicons-halflings-regular.ttf') format('truetype'), url('../fonts/glyphicons-halflings-regular.svg#glyphicons_halflingsregular') format('svg');
}
```

Not sure if that matters. But I thought I would point it out. I'm not worried about old versions of IE on this app anyway but someone out there might be.

**Comment by hollandben** [comment]
I don't think the long run-on should make a difference (crosses fingers). I did add a [test case](https://github.com/hollandben/grunt-cache-bust/pull/130/files#diff-b3bd09fea66ce2d19d2d650fc36e4b56R59) for the `?#iefix` but that was only when renaming the file (my bad). Will reopen this and fix that. Thanks

**Comment by hollandben** [comment]
This looks to be fixed in `v0.5.0`
